### PR TITLE
tools/clang: Remove remaining host clang-format usage (+docs update)

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: bazel/README.md [hunk 1]
```diff
@@ -930,60 +930,8 @@ TEST_TMPDIR=/tmp tools/gen_compilation_database.py
...
-
-Alternatively, if you are a non-root user, you can use a bin dir and add that to `$PATH`
-
-```shell
-mkdir bin
-mv clang-format-ci bin/clang-format
-export PATH=$PATH:$PWD/bin/
-```
-
 Once this is set up, you can run clang-format without docker:
```
Comment (adisuissa) [inline bazel/README.md:935]: Can you consider rephrasing this sentence - not sure what's needs to be set up.

**Review by adisuissa — COMMENTED** [review]
Thanks!
Left a minor comment.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Comment by phlax** [comment]
pushing past (resolved) flake

**Pull request merged**